integration/image/delta_test.go

```diff
@@ -0,0 +1,214 @@
+package image
+
+import (
+	"bufio"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"regexp"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/docker/cli/cli/command"
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/integration-cli/daemon"
+	"github.com/docker/docker/integration-cli/registry"
+)
+
+const registryURI = "127.0.0.1:5000"
+
+// PATH=$PATH:`pwd`/balena-engine TESTDIRS="integration/image" TESTFLAGS="-test.run Delta" hack/make.sh test-integration
+func TestDelta(t *testing.T) {
+	type testCase struct {
+		desc           string
+		base           string
+		target         string
+		delta          string
+		expectedImages []string
+	}
+
+	for _, c := range []testCase{
+		{
+			desc:           "busybox images",
+			base:           "busybox:1.24",
+			target:         "busybox:1.29",
+			delta:          fmt.Sprintf("%s/delta-test:1.24-1.29", registryURI),
+			expectedImages: []string{"busybox:1.24", fmt.Sprintf("%s/delta-test:1.24-1.29", registryURI)},
+		},
+	} {
+		t.Run(c.desc, func(t *testing.T) {
+			c := c
+
+			var (
+				err error
+				rc  io.ReadCloser
+			)
+
+			d := daemon.New(t, "", "balena-engine-daemon", daemon.Config{})
```

Comment:
> For example, from lines 52 to 60 (including daemon.New and and d.Start), maybe that block could be a separate StartDaemon function (maybe).

I was considering that as well but are then not just recreating another pkg similar to https://github.com/balena-os/balena-engine/tree/17.12-resin/integration-cli/daemon
and I also didn't want to add code there since:
> The legacy test suite integration-cli/ is deprecated. No new tests will be added to this suite. Any tests in this suite which require updates should be ported to either the unit test suite or the new API integration test suite. (from: https://github.com/balena-os/balena-engine/blob/17.12-resin/TESTING.md)

Reply:
> Is desirable for a new daemon instance to be created and started for each test case of the for loop?

Like I commented below somewhere I will try to use `daemon.ConfigReload` which would mean we don't have to restart  and could just `defer` the stop (to ensure gc of the daemon in case of failure)

> By the way, the other integration tests that are being skipped, was each test typically creating, starting and stopping their own daemons, rather than sharing the daemon?

The container restart test works for me (just removing the skip at the top) and they do the same procedure of starting/stopping/starting the test daemon